oauthlib/oauth1/rfc5849/signature.py

```diff
@@ -173,7 +173,7 @@ def base_string_uri(uri: str, host: str = None) -> str:
     if ':' in netloc:
```

Comment:
Instead of projecting the urlparse object onto a tuple, I think it would be better to read it's attributes, specifically the port attribute.

It correctly unpacks ipv6:

```
>>> o = urlparse("https://[123:db8:fd00:1000::5]:13000")
>>> o.port
13000
>>> o.hostname
'123:db8:fd00:1000::5'
```